src/doc/unstable-book/src/language-features/loop-break-value.md

```diff
@@ -4,7 +4,86 @@ The tracking issue for this feature is: [#37339]
 
 [#37339]: https://github.com/rust-lang/rust/issues/37339
 
+Documentation to be appended to section G of the book.
+
 ------------------------
 
+### Loops as expressions
+
+Like most things in Rust, loops are expressions; for example, the following is perfectly legal,
+if rather useless:
+
+```rust
+let result = for n in 1..4 {
+    println!("Hello, {}", n);
+};
+assert_eq!(result, ());
+```
```

Comment:
I don't find this example very helpful in understanding the future. The return value of `for` is a nice curiousity but I don't see much didatic value to this.